src/routes/Routes.tsx

```diff
@@ -14,25 +17,49 @@ import {
 import LoginRedirect from "./components/LoginRedirect/LoginRedirect";
 import NotFound from "./components/NotFound/NotFound";
 import RestrictAccess from "./components/RestrictAccess/RestrictAccess";
+import { LayoutsEnum } from "./constants/route.layouts";
 import { PRIVATE_LIST, PUBLIC_LIST } from "./routes.config";
 
+/**
+ * All Layouts
+ */
+const LAYOUTS = {
+  [LayoutsEnum.HEADER_LAYOUT]: HeaderLayout,
+  [LayoutsEnum.SIDEBAR_LAYOUT]: SidebarLayout,
+  [LayoutsEnum.BLANK_LAYOUT]: BlankLayout,
+};
+
 /**
  * Change the default layout to:
  * - HeaderLayout
  * - SidebarLayout
  */
-const DefaultLayout = HeaderLayout;
+const DefaultLayoutEnum = LayoutsEnum.HEADER_LAYOUT;
 
 const Routes = () => {
   const { isAuthenticated } = useAppSelector(state => ({
     isAuthenticated: state.auth?.isAuthenticated,
   }));
 
+  const location = useLocation();
+  const layoutRef = useRef<ElementType>(LAYOUTS[DefaultLayoutEnum]);
+
+  const matchedRoute: RouteItemDef | undefined = useMemo(
```

Comment:
we could memoize the flatten and make it a dictionary for O(1) access later
```
const flattenedPaths: { [key: string]: RouteItemDef } = useMemo(
    () =>
      Object.assign(
        {},
        ...flatMap(
          [...PUBLIC_LIST, ...PRIVATE_LIST],
          (route: RouteItemDef) => ({
            [route.path.toString()]: route,
          })
        )
      ),
    []
  );

  const matchedRoute: RouteItemDef | undefined = useMemo(
    () => flattenedPaths[location.pathname],
    [location.pathname, flattenedPaths]
  );
```

Reply:
But this won't solve the issue that we tried. By flatting this way we can't get the nested routes that all we need to determine the layout of nested routes. I can do it this way 
```
const flattenedPaths: { [key: string]: RouteItemDef } = useMemo(
    () =>
      Object.assign(
        {},
        ...flatten([...PUBLIC_LIST, ...PRIVATE_LIST]).map(
          (route: RouteItemDef) => ({
            [route.path.toString()]: route,
          })
        )
      ),
    []
  );
```
But does it make that much difference with my current implementation?

Reply:
theoritically flatMap vs flatten.map should result in O(n) vs O(2n)

Reply:
> By flatting this way we can't get the nested routes that all we need to determine the layout of nested routes
I see, then in fact `flatten or flatmap` alone  won't work, we need to create a custom flatten function  for `RouteItemDef` as well to flatten all routes

Reply:
Here `flatten` is our custom function for nested routes

Reply:
okay, i thought it's a function of lodash. hmm, it's used in other places for other purposes as well. Modifying it to return a map need more testing.